StRoot/StEvent/StFcsHit.cxx

```diff
@@ -139,3 +141,26 @@ void StFcsHit::print(Option_t *option) const {
     }
     cout << endl;
 }
+
+void StFcsHit::addGeantTrack(const unsigned int id, float e){    
+    // auto cmp = [](auto t){ return t.first == id; }; //c++20
+    auto cmp = [id](decltype(mGeantTracks)::value_type t){ return t.first == id; };
+    auto trk = find_if(begin(mGeantTracks), end(mGeantTracks), cmp);
```

Comment:
A reverse iteration through the vector should be optimal, because (mostly) the hits from a single track will be presented in sequence before the hits from the next track are processed.

Reply:
auto trk = find_if(rbegin(mGeantTracks), rend(mGeantTracks), cmp);
    if(trk == rend(mGeantTracks)){...}

Will this make the auto to be reverse iterator automatically? Or do I need do something like 

  std::vector<pair<unsigned int,float>::reverse_iterator trk= find_if(...)

Reply:
Looks like

    auto trk = find_if(mGeantTracks.rbegin(), mGeantTracks.rend(), cmp);
    if(trk == mGeantTracks.rend()){
      mGeantTracks.push_back(make_pair(id, e));
    }else{
      trk->second += e;
    }

works.

Reply:
Google says begin(x) & end(x) were added by c++11, while rbegin(x), rend(x) etc were added at c++14.